### ISSUE TITLE
validate session method

### ISSUE THREAD

**Issue by UXDart** [bluesky-social/atproto#1437]
**Is your feature request related to a problem? Please describe.**

if I want to add extra functionality, without creating a new repo. which is the best way to validate the current signed in user?

**Describe the solution you'd like**

I would call getSession with jwt using bsky domain. that would be secure, at least for this domain?

I want to save extra info for current user, one option is also to call putPreferences... don't like a lot the way it does the "get and save" every time, instead of an "updatePreferences" instead. I think I have two requests maybe.

Thanks

**Comment by UXDart** [comment]
I see a test that "updates and removes" preferences... but just did a test, and putPreferences call is doing an override of all preferences. it is a bug or a feature? thanks

**Comment by devinivy** [comment]
Yes, you can get the currently signed-in user by hitting the `com.atproto.server.getSession` endpoint with the JWT you received during login.  You should send the request to the user's PDS, which is currently `bsky.social` for users in production.

And yes, all preferences are always updated together.  Our `BskyAgent` in the API client `@atproto/api` has some utilities to facilitate updating individual preferences, which you can find right around here in the codebase https://github.com/bluesky-social/atproto/blob/2fa9088639c1e17e49e524bb8fa2db2d4e9680a7/packages/api/src/bsky-agent.ts#L334
